### PR TITLE
Improve error messaging for missing fzf command

### PR DESCRIPTION
**Pull request by ChrisPenner**
## Overview

Some folks requested better error handling for fzf commands lately.

If `fzf` is missing from the user's path and they run a command which would invoke it, we should inform them of the problem, and also print the help message of the command.

Example outputs:

```
.> view
I couldn't find the `fzf` executable on your path, consider installing `fzf` to enable fuzzy searching.

  view
  `view foo` prints the definition of `foo`. `view` without arguments invokes a search to select
  definitions to view, which requires that `fzf` can be found within your PATH.

.> cd
I couldn't find the `fzf` executable on your path, consider installing `fzf` to enable fuzzy searching.

  namespace (or cd, j)
  Moves your perspective to a different namespace.
  `namespace foo.bar`   descends into foo.bar from the current namespace.
  `namespace .cat.dog`  sets the current namespace to the abolute namespace .cat.dog.
  `namespace ..`        moves to the parent of the current namespace. E.g. moves from '.cat.dog' to
                        '.cat'
  `namespace`           invokes a search to select which namespace to move to, which requires that
                        `fzf` can be found within your PATH.

.> display
I couldn't find the `fzf` executable on your path, consider installing `fzf` to enable fuzzy searching.

  `display foo` prints a rendered version of the term `foo`. `display` without arguments invokes a
  search to select a definition to display, which requires that `fzf` can be found within your
  PATH.
```

## Implementation notes

* Adds a new Output for printing the help message of a given input pattern.
* Prints the appropriate help message if FZF fails due to missing executable.

### CODE REVIEW

**Comment by ceedubs** [comment]
I'll let someone more familiar with the codebase review the code changes, but here is a review of the error messages.

This seems like a big improvement. Thank you @ChrisPenner!

One minor nit: For `view` and `display`, at least as the messages appear in your PR description, it doesn't seem that clear to me that "`fzf` must be found on your PATH." applies only to the no-arg version. For `cd` it is clear.

**Comment by ceedubs** [comment]
The error messages look great to me 🏆.

As I mentioned, I'll let someone else review the code since I don't know what I'm doing when it comes to that.

File: unison-cli/src/Unison/CommandLine/InputPatterns.hs [hunk 1]
```diff
@@ -300,7 +300,11 @@ view =
     "view"
     []
     [(ZeroPlus, definitionQueryArg)]
-    "`view foo` prints the definition of `foo`."
+    ( P.lines
+        [ "`view foo` prints the definition of `foo`.",
+          "`view` without arguments invokes a search to select definitions to view, which requires that `fzf` can be found within your PATH."
```
Comment (rlmark) [inline unison-cli/src/Unison/CommandLine/InputPatterns.hs:305]: Love it!

**Review by rlmark — APPROVED** [review]
I like it! I think it helps people understand what circumstances `fzf` is needed in.

**Comment by aryairani** [comment]
@ChrisPenner Would you go one step further with a little more context info in the message, e.g.:

```
.> view
`view` with no arguments activates fuzzy searching, which requires `fzf` installed, but I can't find
it on your path. Consider installing it, or provide one or more arguments to specify what to view:
```

It doesn't have to be exactly that, but I just imagine someone who typed `view` without knowing what they were doing, nor what they were trying to do, and now they're being told something unexpected about missing software, and "Help." 😅  I want to manage the surprise factor of the text as much as possible.

I think it's tough to find the right balance 🤞😅 to support both the "I don't want to think" group and "I don't want to read" group, but I'd lean slightly towards supporting the former.\</soapbox>

**Review by aryairani — APPROVED** [review]
Approved in spirit!  I left a suggestion for an even more hand-holding-ey message.

**Comment by hojberg** [comment]
Should we mention what fzf is?

**Comment by aryairani** [comment]
> Should we mention what fzf is?

@hojberg Hmm... I think it's clear from a message like that that fzf is a thing that 1) can be installed and 2) supports fuzzy searching.  Do you think they should know more than that?

**Comment by ChrisPenner** [comment]
@aryairani 
Currently I have the messages split up:

1. An `Output` which simply outputs the help text for a command
2. A message indicating that `fzf` isn't installed.

The message

> `view` with no arguments activates fuzzy searching, which requires `fzf` installed, but I can't find
it on your path.

Requires mixing the two together (help message for `view`, and knowledge that `fzf` wasn't found) and this makes the code a bit messier and less re-usable, is there some middle-of-the-road you'd be okay settling on?

If you'd like I can swap the ordering of the messages around so it view like this:

```
  view
  `view foo` prints the definition of `foo`. `view` without arguments invokes a search to select
  definitions to view, which requires that `fzf` can be found within your PATH.

I couldn't find the `fzf` executable on your path, consider installing `fzf` to enable fuzzy searching.
```

**Comment by hojberg** [comment]
@aryairani hmmm, maybe we could just link to the library or something?

**Review by mitchellwrosen — APPROVED** [review]

**Review by ceedubs — APPROVED** [review]
Okay there are 4 approvals and I think that this is pretty clearly an improvement over what was there before. I'm going to go ahead and merge and further refinement can happen in a separate PR if someone has the time and feels motivated 😊.

**Comment by aryairani** [comment]
> If you'd like I can swap the ordering of the messages around so it view like this:
> 
> ```
>   view
>   `view foo` prints the definition of `foo`. `view` without arguments invokes a search to select
>   definitions to view, which requires that `fzf` can be found within your PATH.
> 
> I couldn't find the `fzf` executable on your path, consider installing `fzf` to enable fuzzy searching.
> ```

@ChrisPenner Yeah I think that would do the trick.

**Pull request merged**